File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
       unordered_map<int,int>m;
        int sum = 0;
        m[sum] = 1;
        int res = 0;
        for(int i = 0; i<nums.size(); i++){
            sum += nums[i];
            if(m[sum - k]>0){
                res += m[sum - k];
            }
            m[sum]++;
        }
        return res;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    
    let sum = 0;
    let map = new Map();
    map.set(sum,1);
    let maxSumArr = 0;
    nums.map(n => {
        sum += n;
        if(map.has(sum - k)){
            maxSumArr+=map.get(sum-k);
        }
        map.set(sum,map.has(sum)?map.get(sum)+1:1);
    })
    return maxSumArr;
};
